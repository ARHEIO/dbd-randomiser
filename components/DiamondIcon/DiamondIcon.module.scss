.diamondIconName {
  min-height: 36px;
  text-align: center;
}

.diamondIconIcon {
  margin-top: 20px;
  border: 8px solid #000000;
  width: 100px;
  height: 100px;
  transform: rotateY(0deg) rotate(45.5deg);
  background-color: rgb(172, 62, 227);

  img {
    transform: rotateY(0deg) rotate(-45.5deg);
  }

  &::after {
    display: block;
    content: '';
  }
}
