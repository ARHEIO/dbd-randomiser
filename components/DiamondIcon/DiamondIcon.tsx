import { Rarity } from '@stephenpoole/deadbydaylight';
import React, { ReactElement } from 'react';
import { getRankColour } from '../../src/utils/getRankColour';
import styles from './DiamondIcon.module.scss';

// TODO: Find a way to have a placeholder spinner component for the image

export const DiamondIcon: React.VFC<{name: string; icon: string; rarity: number}> = ({name, icon, rarity}): ReactElement => {

  return (
    <div>
      <p className={styles.diamondIconName}>{name}</p>
      <div className={styles.diamondIconIcon} style={{ backgroundColor: getRankColour(rarity) }}>
        <img alt={name} src={`https://assets.arhe.io/dbd-randomiser/shitty/${icon}.png`} />
      </div>
    </div>
  );
};
