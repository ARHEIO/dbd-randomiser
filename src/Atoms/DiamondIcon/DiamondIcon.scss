.diamond-icon {
  &_name {
    min-height: 36px;
    text-align: center;
  }

  &_icon {
    margin-top: 20px;
    border: 8px solid #000000;
    width: 58%;
    transform: rotateY(0deg) rotate(45.5deg);

    img {
      transform: rotateY(0deg) rotate(-45.5deg);
    }

    &::after {
      display: block;
      content: '';
    }
  }
}
