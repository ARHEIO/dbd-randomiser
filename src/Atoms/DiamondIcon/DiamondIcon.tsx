import React, { ReactElement } from 'react';
import './DiamondIcon.scss';
import { BasicObject } from '../../Services/generator/models';
import colourConfig from '../color-config';
import getAssetLocation from '../../Services/utils';

// TODO: Find a way to have a placeholder spinner component for the image

const DiamondIcon = (props: BasicObject): ReactElement => {
  const { name, rank, icon } = props;
  const backgroundcolor = rank ? colourConfig[`${rank}`] : '#61513C';

  return (
    <div className="diamond-icon">
      <p className="diamond-icon_name">{name}</p>
      <div className="diamond-icon_icon" style={{ backgroundColor: backgroundcolor }}>
        <img alt={name} src={getAssetLocation(icon)} />
      </div>
    </div>
  );
};

export default DiamondIcon;
