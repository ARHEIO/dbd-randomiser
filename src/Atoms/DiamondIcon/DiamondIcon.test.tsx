import React from 'react';

import {
  render,
} from '@testing-library/react';
import pretty from 'pretty';
import DiamondIcon from './DiamondIcon';

describe('Diamond Icon', () => {
  it('should render when the default object is passed', () => {
    const { container } = render(<DiamondIcon name="Adam" rank={1} icon="" />);
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('should render even when rank is null', () => {
    const { container } = render(<DiamondIcon name="Adam" rank={null} icon="" />);
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
