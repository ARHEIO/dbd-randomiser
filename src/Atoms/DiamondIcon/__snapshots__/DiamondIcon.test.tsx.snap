// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diamond Icon should render even when rank is null 1`] = `
"<div class=\\"diamond-icon\\">
  <p class=\\"diamond-icon_name\\">Adam</p>
  <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(97, 81, 60);\\"><img alt=\\"Adam\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
</div>"
`;

exports[`Diamond Icon should render when the default object is passed 1`] = `
"<div class=\\"diamond-icon\\">
  <p class=\\"diamond-icon_name\\">Adam</p>
  <div class=\\"diamond-icon_icon\\" style=\\"background-color: rgb(171, 113, 60);\\"><img alt=\\"Adam\\" src=\\"https://dqr3pglopijar.cloudfront.net/\\"></div>
</div>"
`;
